src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestCallback.cs

```diff
@@ -306,7 +306,10 @@ private static void OnRequestError(WinHttpRequestState state, Interop.WinHttp.WI
             Debug.Assert(state != null, "OnRequestError: state is null");
 
             Exception innerException = WinHttpException.CreateExceptionUsingError(unchecked((int)asyncResult.dwError));
-
+            
+            Debug.Assert(unchecked((int)asyncResult.dwError) != Interop.WinHttp.ERROR_INSUFFICIENT_BUFFER, "5 called function ID: " + unchecked((uint)asyncResult.dwResult.ToInt32()));
```

Comment:
If this fails, isn't it 100% certain that asyncResult.dwError is exactly the one we're checking in the assert?